code/modules/mob/dead/new_player/new_player.dm

```diff
@@ -426,51 +426,83 @@
 	for(var/datum/job/job in SSjob.occupations)
 		if(job && IsJobUnavailable(job.title, TRUE) == JOB_AVAILABLE)
 			available_job_count++;
+			break;
 
-	for(var/datum/job/prioritized_job in SSjob.prioritized_jobs)
-		if(prioritized_job.current_positions >= prioritized_job.total_positions)
-			SSjob.prioritized_jobs -= prioritized_job
-
-	if(length(SSjob.prioritized_jobs))
-		dat += "<div class='notice red'>The station has flagged these jobs as high priority:<br>"
-		var/amt = length(SSjob.prioritized_jobs)
-		var/amt_count
-		for(var/datum/job/a in SSjob.prioritized_jobs)
-			amt_count++
-			if(amt_count != amt) // checks for the last job added.
-				dat += " [a.title], "
-			else
-				dat += " [a.title]. </div>"
+	if(!available_job_count)
+		dat += "<div class='notice red'>There are currently no open positions!</div>"
 
-	dat += "<div class='clearBoth'>Choose from the following open positions:</div><br>"
-	dat += "<div class='jobs'><div class='jobsColumn'>"
-	var/job_count = 0
-	for(var/datum/job/job in SSjob.occupations)
-		if(job && IsJobUnavailable(job.title, TRUE) == JOB_AVAILABLE)
-			job_count++;
-			if (job_count > round(available_job_count / 2))
-				dat += "</div><div class='jobsColumn'>"
-			var/position_class = "otherPosition"
-			if (job.title in GLOB.command_positions)
-				position_class = "commandPosition"
-			dat += "<a class='[position_class]' href='byond://?src=[REF(src)];SelectedJob=[job.title]'>[job.title] ([job.current_positions])</a><br>"
-	if(!job_count) //if there's nowhere to go, overflow opens up.
+	else
+
+	// if(length(SSjob.prioritized_jobs))
+	// 	dat += "<div class='notice red'>The station has flagged these jobs as high priority:<br>"
+	// 	for(var/datum/job/a in SSjob.prioritized_jobs)
+	// 		dat += " [a.title], "
+	// 	dat += "</div>"
+
+		dat += "<div class='clearBoth'>Choose from the following open positions:</div><br>"
+		var/list/categorizedJobs = list(
+			"Command" = list(jobs = list(), titles = GLOB.command_positions, color = "#aac1ee"),
+			"Engineering" = list(jobs = list(), titles = GLOB.engineering_positions, color = "#ffd699"),
+			"Supply" = list(jobs = list(), titles = GLOB.supply_positions, color = "#ead4ae"),
+			"Miscellaneous" = list(jobs = list(), titles = list(), color = "#ffffff", colBreak = TRUE),
+			"Synthetic" = list(jobs = list(), titles = GLOB.nonhuman_positions, color = "#ccffcc"),
+			"Service" = list(jobs = list(), titles = GLOB.civilian_positions, color = "#cccccc"),
+			"Medical" = list(jobs = list(), titles = GLOB.medical_positions, color = "#99ffe6", colBreak = TRUE),
+			"Science" = list(jobs = list(), titles = GLOB.science_positions, color = "#e6b3e6"),
+			"Security" = list(jobs = list(), titles = GLOB.security_positions, color = "#ff9999"),
+		)
 		for(var/datum/job/job in SSjob.occupations)
-			if(job.title != SSjob.overflow_role)
+			if(job && IsJobUnavailable(job.title, TRUE) == JOB_AVAILABLE)

```

Comment:
This whole block is just bleh. Every time the panel is opened, it loops through every job to create several lists that are essentially, the same as the existing global job positions lists. Just use those lists. Or maybe modify the code so you don't need to grab the actual job datum instead of just the name.

Reply:
This idea is a bit more fruity: Since the job datums are stored only to access `job.title` and `job.current_positions`, it's possible to do away with that whole process by using the titles from each global jobs lists (they're the same) and directly accessing `SSjob.occupation<job>.current_positions`. It may be a bit squirrely to retrieve however, haven't had a proper look.